docs/mirror_configuration.md

```diff
@@ -1,272 +1,408 @@
 ```
 
-## diff-append-epoch
+### `download-mirror`
 
-The diff append epoch is a boolean (true/false) setting that indicates if the diff-file should be appended with the current epoch time.
-This can be used to track diffs over time so the diff file doesn't get cobbered each run.  It is only used when diff-file is used.
+Download package release files from an alternative repository server.
 
-Example:
+:Type: URL
+:Default: none
 
-```ini
-[mirror]
-diff-append-epoch = true
-```
+By default, Bandersnatch downloads packages from the URL supplied in the master server's JSON response. Setting this option to a repository URL will try to download release files from that repository first, and fallback to the URL supplied by the master server if that is unsuccessful (unable to get content or checksum mismatch).
 
-## compare-method
+This is useful to sync most of the files from an existing, nearby mirror - for example, when creating a new mirror identical to an existing one for the purpose of load sharing.
 
-The compare method is used to set how to compare an existing file with upstream file to determine whether a download is required:
+### `download-mirror-no-fallback`
 
-- hash: this is the default which reads local file content and computes hashes (currently sha256sum), it is reliable but sometimes slower;
-- stat: use file size and change time to compare, which is named after the stat() syscall, this avoids retrieving the full file content thus reducing some io workloads.
+Disable the fallback behavior for [](./mirror_configuration.md#download-mirror).
 
-Example:
+:Type: boolean
+:Default: false
 
-```ini
-[mirror]
-compare-method = hash
+When set to `true`, Bandersnatch only downloads package distribution artifacts from the repository set in [](./mirror_configuration.md#download-mirror) and ignores file URLs received from the [](./mirror_configuration.md#master) server.
+
+```{warning}
+This could lead to more failures than expected and is not recommended for most scenarios.
 ```
 
-## proxy
+%
+% processing and miscellaneous options
+%
 
-The proxy is used only when requesting master server, eg. downloading index or package file from pypi.org.
-The proxy value will be passed to aiohttp as proxy parameter, like `aiohttp.get(link, proxy=yourproxy)`,
-check the aioproxy manual for more details: <https://docs.aiohttp.org/en/stable/client_advanced.html#proxy-support>
+### `cleanup`
 
-Example:
+Enable cleanup of legacy simple directories with non-normalized names.
 
-```ini
-[mirror]
-proxy=http://myproxy.com
+:Type: boolean
+:Default: false
+
+Bandersnatch versions prior to 4.0 used directories with non-normalized package names for compatability with older versions of pip. Enabling this option checks for and removes these directories.
+
+```{seealso}
+[Python Packaging User Guide - Names and Normalization](https://packaging.python.org/en/latest/specifications/name-normalization/)
```

Comment:
I should prob remove this one day.

Reply:
Would it make sense to go ahead and indicate that it's deprecated in the docs? (Though I suppose technically that requires a version bump even if its just the documented status that changes?)

Reply:
Yeah, will not hurt.